### ISSUE TITLE
196 Lag and Server Overloaded Messages

### ISSUE THREAD

**Issue by isaaclepes** [PaperMC/Paper#3182]
###  Timings or Profile link:
___We ask that all timings/profiles are a link, not a screenshot. Screenshots inhibit our ability to figure out the real cause of the issue.___

I am fairly new to hosting a server, how do I pull this data?

### Description of issue:
___If applicable___

Was using Paper 184 and upgraded to 196 today.  After shutting down the server, changing out the paper.jar, and starting it back up; the server has been lagging and even with no players on it reports a log almost identical to this every 30-40 seconds:
[20:40:41] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 5698ms or 113 ticks behind

This starts after the server initializes to this point, but not before:
[17:31:26] [Server thread/INFO]: Time elapsed: 5740 ms
[17:31:27] [Server thread/INFO]: Done (98.363s)! For help, type "help"
[17:31:27] [Server thread/INFO]: Starting remote control listener
[17:31:27] [RCON Listener #1/INFO]: RCON running on 10.0.0.12:25575
[17:31:27] [Server thread/INFO]: Timings Reset
[17:31:33] [Server thread/WARN]: Can't keep up! Is the server overloaded? Running 5550ms or 111 ticks behind

### Plugin list:
___All plugins running on the server___
WorldEdit 7.1.0

### bukkit.yml, spigot.yml, paper.yml, server.properties
___Gist/pastebin/hastebin links___

server.properties https://pastebin.com/T6gRJz8b
bukkit.yml  https://pastebin.com/di4DkbA7
spigot.yml  https://pastebin.com/TrAAs1j8
paper.yml  https://pastebin.com/ikJ6EKXz

### Other helpful links
___The more information we receive, the quicker and more effective we can be at finding the solution to the
issue.___

### Paper build number:
___This can be found by running `/version` on your server. `latest` is not a proper version number; we require the output of `/version` so we can properly track down the issue.___

196

**Comment by isaaclepes** [comment]
Switching back to the 184.jar solves the issue.  If I have some time later, I will try each version between until the issue returns.

**Comment by aikar** [comment]
You did not provide a timings report, please provide us one or really nothing can be done here.

**Comment by isaaclepes** [comment]
I did not know how to do a timings report, which is why I asked in my initial post.  After some searching I located a guide and will have a look. (If anyone else needs this:  https://www.spigotmc.org/wiki/how-to-take-a-timings-report/ )

Timings report:  https://timings.aikar.co/?id=22bfce5a11a74b4bb68920b764820502


The server was created last week in a Debian 10 virtual machine on a Dell R710 and was performing great as a 2-person LAN server until updating to 196.  I was able to reduce but not eliminate the lag by quadrupling the CPU resources assigned to the VM, but I do no believe the vm should require 8 vcpu to run a world with no one connected.  RAM usage is low, at least.  (6GB RAM, 60GB free disk space on a SAS RAID 5).  Timings report was created after adding 6 vCPUs, but the issue still presented itself.

**Comment by aikar** [comment]
Sorry missed your 'how do i pull this'.

from timings, it looks like you have a lot of blocks ticks going on, potentially from functions from data packs.

Nothing in our recent changes really could have caused any issues here, as nearly everything improved performance.

The main changes to block ticking was done WAYYYY before build 184....

I'm concerned your data packs are just randomly causing some spurious updates, and it's not really related to paper version.

Can you get multiple timing reports (over 1hour~) span on old version and same on new version?

**Comment by isaaclepes** [comment]
Aikar,

I decided to try to simplify things and moved my world directory causing a new one to be created with the same seed.  No problems on a fresh version of my world, so I guess I will have to sort out what in the world is causing it.  My brother and I have barely started to build in it, so I doubt it was our basic starter houses.  There is a large village near spawn, if that matters.  (Seed 8675309)

**Comment by chickeneer** [comment]
Your timings indicated that you have 768 Repeating Command Blocks. The timings report doesn't say what they are doing - or if they are doing anything at all. But that seems highly suspect to me.

**Comment by isaaclepes** [comment]
I loaded the old world up and used creative/spectator to look around.  I noticed something very off under the surface of the original world;  a lot of blocks were just strange.  Interspersed in the stone were things that shouldn't exist, like waterlogged double-stone slabs, blocks that looked like emerald ore but were actually gravel when mined, and other nonsense.  The F3 screen even said it was emerald, but when broken it was gravel.  I hadn't encountered anything like this on initial exploration/building, so it is probably something from testing out plugins/datapacks that didn't express until I restarted to switch to paper-196.  The newly generated world looks just fine.. those emerald ores are actually gravel as they should be, etc.  I guess plugins aren't as easy to try/remove as deleting them.  I bet buried under the surface are a load of command blocks that will become something else when mined, lol.
